### PR TITLE
docs(svelte-testing-library): update API docs for v5 release

### PR DESCRIPTION
**Pull request by mcous**
This is a documentation companion to testing-library/svelte-testing-library#325. We're introducing a few minor, though technically breaking, changes to the API in the upcoming `v5` release (which should be triggered upon merge of the PR above).

This PR updates the `API` section of the `svelte-testing-library` docs accordingly, with callouts for users of the current `v4`.

CC @yanick

### CODE REVIEW

**Review by yanick — APPROVED** [review]
:+1: Awesome. Let's flip the `draft` status once v5 hits the shelves!

File: docs/svelte-testing-library/api.mdx [hunk 1]
```diff
@@ -4,149 +4,321 @@ title: API
...
+
+#### `component`
+
+The Svelte component instance. See the [Svelte component
+API][svelte-component-api] for more details.
+
+:::tip
+
+Avoid using `component` except to test developer facing API's, like exported
+functions. Instead, prefer of interacting with the DOM. Read [Avoid the Test
```
Comment (mcous) [inline docs/svelte-testing-library/api.mdx:145]: Typo

```suggestion
functions. Instead, interact with the DOM. Read [Avoid the Test
```

File: docs/svelte-testing-library/api.mdx [hunk 2]
```diff
@@ -4,149 +4,321 @@ title: API
...
+User][test-user] by Kent C. Dodds to understand the difference between the **end
+user** and **developer user**.
+
+:::
+
+[test-user]: https://kentcdodds.com/blog/avoid-the-test-user
+
+#### `debug`
+
+Log any passed element, or the `baseElement` by default, using
```
Comment (mcous) [inline docs/svelte-testing-library/api.mdx:155]:
```suggestion
Log an element, or the `baseElement` by default, using
```

File: docs/svelte-testing-library/api.mdx [hunk 3]
```diff
@@ -4,149 +4,321 @@ title: API
...
+[prettyDOM][pretty-dom].
+
+:::tip
+
+If your `baseElement` is the default `document.body`, we recommend using
+[`screen.debug`][screen-debug] rather than the bound `debug`..
```
Comment (mcous) [inline docs/svelte-testing-library/api.mdx:161]:
```suggestion
[`screen.debug`][screen-debug] rather than the bound `debug`.
```

**Review by mcous — COMMENTED** [review]
A few typos

**Review by yanick — APPROVED** [review]

**Comment by mcous** [comment]
@timdeschryver (pinging you because you’ve reviewed my previous STL docs PRs) we just released v5 to our `latest` tag, so these docs are ready to go. I’ve synced up and undrafted the PR accordingly

CC @yanick

File: docs/svelte-testing-library/api.mdx [hunk 4]
```diff
@@ -4,149 +4,306 @@ title: API
...
-📝 If the only option you're passing in is `props`, then you can just pass them
-in directly.
+If you only need to send props to your component, you may pass props directly,
+as long as those props don't share a name with a component option.
 
 ```js
-// With options.
-const view = render(YourComponent, {
-  target: MyTarget,
+// pass props to component
```
Comment (timdeschryver) [inline docs/svelte-testing-library/api.mdx:38]:
```suggestion
// pass props to the component
```

File: docs/svelte-testing-library/api.mdx [hunk 5]
```diff
@@ -4,149 +4,306 @@ title: API
...
+render(YourComponent, {myProp: 'value'})
+
+// pass props and other options to component
```
Comment (timdeschryver) [inline docs/svelte-testing-library/api.mdx:41]:
```suggestion
// pass props and other options to the component
```

File: docs/svelte-testing-library/api.mdx [hunk 6]
```diff
@@ -4,149 +4,306 @@ title: API
...
+:::
+
+#### `component`
+
+The Svelte component instance. See the [Svelte component
+API][svelte-component-api] for more details.
+
+:::tip
+
+Avoid using `component` except to test developer facing API's, like exported
```
Comment (timdeschryver) [inline docs/svelte-testing-library/api.mdx:134]:
```suggestion
Avoid using `component` except to test developer-facing APIs, like exported
```

**Review by timdeschryver — COMMENTED** [review]
I like the admonitions! 🤩
I left a few nits.

**Comment by mcous** [comment]
Thanks for the careful review!

**Comment by mcous** [comment]
@timdeschryver Is there anything blocking merge here? We just got an inbound issue due to improper API usage that this doc update details

**Comment by timdeschryver** [comment]
> @timdeschryver Is there anything blocking merge here? We just got an inbound issue due to improper API usage that this doc update details

Oops 😅 it got off my radar. Thanks for the ping!

**Pull request merged**